### PR TITLE
chore(cmsis-pack): prepare for LVGL9 release

### PR DESCRIPTION
**Pull request by GorgonMeducer**
Help us review this PR! Anyone can [approve it or request changes](https://docs.github.com/en/pull-requests/collaborating-with-pull-requests/reviewing-changes-in-pull-requests/approving-a-pull-request-with-required-reviews).

### Description of the feature or fix

- Creating cmsis-pack for LVGL9 release.
- Other minor updates

### Checkpoints
- Update the [Documentation](https://github.com/lvgl/lvgl/tree/master/docs) if needed.
- Add [Examples](https://github.com/lvgl/lvgl/tree/master/examples) if relevant.
- Add [Tests](https://github.com/lvgl/lvgl/blob/master/tests/README.md) if applicable.
- If you added new options to `lv_conf_template.h` run [lv_conf_internal_gen.py](https://github.com/lvgl/lvgl/blob/master/scripts/lv_conf_internal_gen.py) and update [Kconfig](https://github.com/lvgl/lvgl/blob/master/Kconfig).
- Run `scripts/code-format.py` ([astyle](http://astyle.sourceforge.net/install.html) needs to be installed) and follow the [Code Conventions](https://docs.lvgl.io/master/CODING_STYLE.html)

### CODE REVIEW

**Comment by GorgonMeducer** [comment]
@kisvegabor 
I have created dedicated header files for arm-2d and included them in `lv_draw_sw_helium.h` and `lv_blend_helium.h`.
Could this relief your concerns?

File: README.md [hunk 1]
```diff
@@ -1,6 +1,3 @@
-**IMPORTANT NOTE** The next major version (v9.0.0) is developed in the master branch.
-The last stable version is available in the [release/v8.3](https://github.com/lvgl/lvgl/tree/release/v8.3) branch.
-
 ---
```
Comment (kisvegabor) [inline README.md]:
```suggestion

```

**Review by kisvegabor — COMMENTED** [review]
I think the root of my confusion with Helium/Arm2D is that
1. Now we have Arm2D/Helium to accelerate SW render via intrinsics: I think it's not helium because Arm2D can use anything under the hood. That is we could use the current Arm2D intrinsics on a not-helium MCU too, right?
2. We can have an Arm2D draw unit: I'm not sure if it's planned/needed but if so, it would be also Arm2D.

So I wonder why do we needed the world "helium" in connection with Arm2D. Please let me know if I'm missing something?

**Comment by GorgonMeducer** [comment]
For the concerns you have:
> I think it's not helium because Arm2D can use anything under the hood. That is we could use the current Arm2D intrinsics on a not-helium MCU too, right?

In theory, yes, but I have added macro to ensure that **only when Helium is available in a device, the arm-2d intrinsics would be used.** 

For now, there is no reason to enable arm-2d acceleration for non-helium MCU. It gains no benefits at all.

> We can have an Arm2D draw unit: I'm not sure if it's planned/needed but if so, it would be also Arm2D.

Yes, it is planned. Let me explain it in detail:

- Arm-2D provides acceleration for both Synchronose acceleration and Asyncrhonouse acceleration
    * Helium and Helium-based ACI (Arm Custom Instruction Set) are synchronous acceleration. I use this synchronous acceleration in the draw-sw for blend and transform.
    * Arm also provides an IP called DMA350 which can be used as a skeleton of a 2D-GPU that silicon vendors can expand its 2D processing capability with their extensions. As a 2D-GPU, it offloads CPUs. The Arm-2D Draw Unit is used as the driver for this 2D-GPU. 

As you can see, the one accelerating draw-sw and the Arm-2D Draw Unit play different roles and use different aspects of arm-2d to accelerate LVGL9 in any possible way.

We plan to add the support for DMAC-350 this year and once it is ready, we will implement the Arm-2D DrawUnit and enable the DMAC-350 as a 2D GPU for LVGL.

I hope this answers you questions.

**Comment by kisvegabor** [comment]
Thank you for the clarification. The sync-async part is clear too. 

> In theory, yes, but I have added macro to ensure that only when Helium is available in a device, the arm-2d intrinsics would be used.
For now, there is no reason to enable arm-2d acceleration for non-helium MCU. It gains no benefits at all.

I feel like it would be more future proof the call the Sync version Arm2D as it might happen that it will support other magics besides Helium. What do you think?

**Comment by GorgonMeducer** [comment]
> I wonder why do we needed the world "helium" in connection with Arm2D

For draw-sw, I believe the **only valuable part of arm-2d is the helium acceleration**. I have and will continue to make sure that **only** the helium part is used.

Further more, I would like to use the arm-2d helium part as the default helium implementation of the LVGL intrinsics. This means:

- People can choose some of the LVGL intrinsics and add their own helium implementation
    * no need to remove the arm-2d
    * Just define the intrinsics macros (directly or via `LV_DRAW_SW_HELIUM_CUSTOM_INCLUDE `)

**Comment by GorgonMeducer** [comment]
> I feel like it would be more future proof the call the Sync version Arm2D as it might happen that it will support other magics besides Helium. What do you think?

No problem at all. What do you suggest to do?

**Comment by kisvegabor** [comment]
Can/will Arm2D support Neon or general non-SIMD assembly intrinsics in the future? I so, I wouldn't mention helium at all in folder names and config defines and just call it Arm2D.

**Comment by GorgonMeducer** [comment]
>Can/will Arm2D support Neon or general non-SIMD assembly intrinsics in the future?

- Arm-2D has the plan to add NEON support, but it won't happen this year.

- Arm-2D defines intrincis in low-level, and we use helium to implement those intrinsics.  In other words, arm-2d supports non-SIMD assembly intrinsics in theory and you can use any way to implement those intrinsics in arm-2d. 

I am dividing the macro `LV_USE_DRAW_ARM2D` into two separated ones, i.e. `LV_USE_DRAW_ARM2D_SYNC` and `LV_USE_DRAW_ARM2D_ASYNC`.

File: lvgl.h [hunk 2]
```diff
@@ -16,7 +16,7 @@ extern "C" {
 #define LVGL_VERSION_MAJOR 9
 #define LVGL_VERSION_MINOR 0
 #define LVGL_VERSION_PATCH 0
-#define LVGL_VERSION_INFO "dev"
```
Comment (kisvegabor) [inline lvgl.h:19]: In line with [this comment](https://github.com/lvgl/lvgl/issues/4926#issuecomment-1891789674) please use the `"rc"` tag for now.

Comment (GorgonMeducer) [reply lvgl.h:19]: Done

**Comment by kisvegabor** [comment]
> I am dividing the macro LV_USE_DRAW_ARM2D into two separated ones, i.e. LV_USE_DRAW_ARM2D_SYNC and LV_USE_DRAW_ARM2D_ASYNC.

I think it better shows the what is what. The last commit looks good. I think the last missing thing is to rename helium folders to arm2d.

**Comment by GorgonMeducer** [comment]
OK, no problem. 
I will make the change.

**Comment by GorgonMeducer** [comment]
@kisvegabor Done. 
Please check.

File: lv_conf_template.h [hunk 3]
```diff
@@ -133,8 +133,8 @@
     #endif
 #endif
 
-/* Use Arm-2D on Cortex-M based devices. Please only enable it for Helium Powered devices for now */
-#define LV_USE_DRAW_ARM2D 0
+/* Use Arm-2D to accelerate the sw render */
+#define LV_USE_DRAW_ARM2D_SYNC 0
```
Comment (kisvegabor) [inline lv_conf_template.h]: If it's used only with the SW render please move it to  e.g. line 103.

**Review by kisvegabor — COMMENTED** [review]
What happens if `LV_USE_DRAW_ARM2D_SYNC` is enabled but `LV_DRAW_SW_ASM_HELIUM` is not?

**Comment by GorgonMeducer** [comment]
No Helium acceleration would be added to draw-sw

File: lv_conf_template.h (hunk 3 above)
Comment (GorgonMeducer) [reply lv_conf_template.h]: OK, no problem

File: library.properties [hunk 4]
```diff
@@ -1,5 +1,5 @@
 name=lvgl
-version=9.0.0-dev
+version=9.0.0
```
Comment (kisvegabor) [inline library.properties]:
```suggestion
version=9.0.0-rc
```

File: library.json [hunk 5]
```diff
@@ -1,6 +1,6 @@
 {
 	"name": "lvgl",
-	"version": "9.0.0-dev",
+	"version": "9.0.0",
```
Comment (kisvegabor) [inline library.json]:
```suggestion
	"version": "9.0.0-rc",
```

File: lv_conf_template.h [hunk 6]
```diff
@@ -1,6 +1,6 @@
 /**
  * @file lv_conf.h
- * Configuration file for v9.0.0-dev
+ * Configuration file for v9.0.0
```
Comment (kisvegabor) [inline lv_conf_template.h]:
```suggestion
 * Configuration file for v9.0.0-rc
```

**Review by kisvegabor — COMMENTED** [review]
Thank you, Gabriel. 

I think it's much cleaner now. Just the last touch with the "-rc" tags.

**Comment by GorgonMeducer** [comment]
You can merge it now as it is updated to the latest changes.

**Review by kisvegabor — APPROVED** [review]

**Comment by kisvegabor** [comment]
Thank you! I'm writing the news letter now :slightly_smiling_face:

**Pull request merged**